cardano-config/src/Cardano/Node/Configuration/Basics.hs

```diff
@@ -0,0 +1,66 @@
+-- | Basic types for configuration
+module Cardano.Node.Configuration.Basics
+  ( -- * FilePaths
+    FilePath (..)
+  , (</>)
+  , RelativeFilePath (..)
+  , anchorRelativePath
+
+    -- * Defaults
+  , Override (..)
+  , (.:=)
+  ) where
+
+import Control.Applicative ((<|>))
+import Data.Aeson
+import Data.Aeson.Types (Parser)
+import Data.Default
+import Data.String (IsString)
+import GHC.Generics (Generic)
+import GHC.TypeLits (Symbol)
+import qualified System.FilePath as F
+import Prelude hiding (FilePath)
+import qualified Prelude as P
+
+-- | A filepath annotated with the intended use for it
+newtype FilePath (s :: Symbol) = FilePath {getFilePath :: P.FilePath}
+  deriving (Generic, Show)
+  deriving newtype (FromJSON, IsString)
+
+(</>) :: FilePath a -> FilePath b -> FilePath c
+FilePath a </> FilePath b = FilePath (a F.</> b)
+
+-- | A FilePath that requires a mount point
+newtype RelativeFilePath (s :: Symbol)
+  = RelativeFilePath {relativeFilePath :: FilePath s}
+  deriving (Generic, Show)
+
+anchorRelativePath :: FilePath a -> RelativeFilePath b -> FilePath b
+anchorRelativePath fp1 (RelativeFilePath fp2) = fp1 </> fp2
```

Comment:
There's a footgun here if someone puts an absolute path into `fp2`: `fp1` will be ignored https://hackage-content.haskell.org/package/filepath-1.5.4.0/docs/System-FilePath.html#v:-60--47--62-

I don't know if there's an easy way to protect against, which is portable. Checking for relative paths requires `IO` access afair. But a haddock with a warning would be nice. 😉

Reply:
Done